mapie/regression/regression.py

```diff
@@ -708,74 +540,22 @@ def predict(
-
-        alpha_np = cast(NDArray, alpha)
-        check_alpha_and_n_samples(alpha_np, n)
-
-        if self.method in self.no_agg_methods_ \
-                or self.cv in self.no_agg_cv_:
-            y_pred_multi_low = y_pred[:, np.newaxis]
-            y_pred_multi_up = y_pred[:, np.newaxis]
-        else:
-            y_pred_multi = self._pred_multi(X)
-
-            if self.method == "minmax":
-                y_pred_multi_low = np.min(y_pred_multi, axis=1, keepdims=True)
-                y_pred_multi_up = np.max(y_pred_multi, axis=1, keepdims=True)
-            else:
-                y_pred_multi_low = y_pred_multi
-                y_pred_multi_up = y_pred_multi
-
-            if ensemble:
-                y_pred = aggregate_all(self.agg_function, y_pred_multi)
-
-        # compute distributions of lower and upper bounds
-        if self.conformity_score_function_.sym:
-            conformity_scores_low = -self.conformity_scores_
-            conformity_scores_up = self.conformity_scores_
-        else:
-            conformity_scores_low = self.conformity_scores_
-            conformity_scores_up = self.conformity_scores_
-            alpha_np = alpha_np / 2
-
-        lower_bounds = (
-            self.conformity_score_function_.get_estimation_distribution(
-                y_pred_multi_low, conformity_scores_low
-            )
-        )
-        upper_bounds = (
-            self.conformity_score_function_.get_estimation_distribution(
-                y_pred_multi_up, conformity_scores_up
-            )
-        )
-
-        # get desired confidence intervals according to alpha
-        y_pred_low = np.column_stack(
-            [
-                np_nanquantile(
-                    lower_bounds.astype(float),
-                    _alpha,
-                    axis=1,
-                    method="lower",
-                )
-                for _alpha in alpha_np
-            ]
-        )
-        y_pred_up = np.column_stack(
-            [
-                np_nanquantile(
-                    upper_bounds.astype(float),
-                    1 - _alpha,
-                    axis=1,
-                    method="higher",
+        if not (alpha is None):
+            n = len(self.conformity_scores_)
+            alpha_np = cast(NDArray, alpha)
+            check_alpha_and_n_samples(alpha_np, n)
+
+            y_pred, bound_low, bound_up = \
+                self.conformity_score_function_.get_bounds(
+                    X,
+                    self.estimator_,
+                    self.conformity_scores_,
+                    alpha_np,
+                    ensemble,
+                    self.method
                 )
-                for _alpha in alpha_np
-            ]
-        )
+            return y_pred, np.stack([bound_low, bound_up], axis=1)
 
-        return y_pred, np.stack([y_pred_low, y_pred_up], axis=1)
+        else:
+            y_pred, _, _ = self.estimator_.predict(X, ensemble)
+            return y_pred
```

Comment:
UPD: it should be moved before.

Reply:
SUGGESTION: Is it possible to force `predict` to return only `y_pred` to avoid calculating y bounds?

Reply:
We could add a parameter in predict to return in a case `y_pred` and in the other `y_pred_multi_low` and `y_pred_multi_up`. But either way the bounds will be calculated : `y_pred` needs the result of `pred_multi(...)`

Reply:
This depends on the value of `ensemble`, doesn't it?

Reply:
In any case, `MapieRegressor` has a `single_estimator_`. Thus if we want only one prediction, we want call `single_estimator_.predict` only.

Reply:
I think I can make it work with `ensemble`.

Reply:
I've considered several options and I don't see how it would be possible to return only `y_pred` in some cases without adding a parameter to the function.
(`ensemble` is not the answer)
Do you see one ?

Reply:
You can add a parameter to the function that make possible to return only `y_pred` in some cases.

Reply:
Done